public/app/features/datasources/components/EditDataSource.tsx

```diff
@@ -204,11 +206,13 @@ export function EditDataSourceView({
       {extensions.map((extension) => {
         const Component = extension.component as React.ComponentType<{
           context: PluginExtensionDataSourceConfigContext<DataSourceJsonData>;
+          showBlockFormFields?: boolean;
```

Comment:
Makes me think whether this should be in the `PluginExtensionDataSourceConfigContext` cc @leventebalogh

Reply:
Hmm, yea, @jarben is correct, everything context specific (information related to where the extension is going to be displayed) should go into the `context` object. 

On the other I don't think this new prop is necessary as _it looks specific for certain type of extensions_. I think the extensions can already access this info from the context (`dataSource.type`) and the feature toggles through`@grafana/runtime`, and they can assemble the conditionals based on their side.